src/internet_identity/src/vc_mvp.rs

```diff
@@ -0,0 +1,247 @@
+            .expect("failed getting signing_input");
+        let msg_hash = vc_signing_input_hash(&signing_input);
+        let maybe_sig = get_signature(cert_assets, sigs, seed, msg_hash);
+        let rp_sig = if let Some(sig) = maybe_sig {
+            sig
+        } else {
+            return GetIdAliasResponse::NoSuchCredentials("rp_sig not found".to_string());
+        };
+        let rp_jws = vc_jwt_to_jws(rp_id_alias_jwt, &canister_sig_pk, &rp_sig)
+            .expect("failed constructing JWS");
+
+        let signing_input = vc_signing_input(issuer_id_alias_jwt, &canister_sig_pk)
+            .expect("failed getting signing_input");
+        let msg_hash = vc_signing_input_hash(&signing_input);
+        let maybe_sig = get_signature(cert_assets, sigs, seed, msg_hash);
+        let issuer_sig = if let Some(sig) = maybe_sig {
+            sig
+        } else {
+            return GetIdAliasResponse::NoSuchCredentials("issuer_sig not found".to_string());
+        };
+        let issuer_jws = vc_jwt_to_jws(issuer_id_alias_jwt, &canister_sig_pk, &issuer_sig)
+            .expect("failed constructing JWS");
+
+        GetIdAliasResponse::Ok(IdAliasCredentials {
+            rp_id_alias_credential: SignedIdAlias {
+                id_alias: id_alias_principal,
+                id_dapp: id_rp,
+                credential_jws: rp_jws,
+            },
+            issuer_id_alias_credential: SignedIdAlias {
+                id_alias: id_alias_principal,
+                id_dapp: id_issuer,
+                credential_jws: issuer_jws,
+            },
+        })
+    })
+}
+
+fn get_signature(
+    cert_assets: &CertifiedAssets,
+    sigs: &SignatureMap,
+    seed: Hash,
+    msg_hash: Hash,
+) -> Option<Vec<u8>> {
+    let certificate = data_certificate().unwrap_or_else(|| {
+        trap("data certificate is only available in query calls");
+    });
+    let witness = sigs.witness(hash::hash_bytes(seed), msg_hash)?;
+
+    let witness_hash = witness.reconstruct();
+    let root_hash = sigs.root_hash();
+    if witness_hash != root_hash {
+        trap(&format!(
+            "internal error: signature map computed an invalid hash tree, witness hash is {}, root hash is {}",
+            hex::encode(witness_hash),
+            hex::encode(root_hash)
+        ));
+    }
+
+    let tree = ic_certified_map::fork(
+        HashTree::Pruned(cert_assets.root_hash()),
+        ic_certified_map::labeled(LABEL_SIG, witness),
+    );
+
+    #[derive(Serialize)]
+    struct Sig<'a> {
+        certificate: ByteBuf,
+        tree: HashTree<'a>,
+    }
+
+    let sig = Sig {
+        certificate: ByteBuf::from(certificate),
+        tree,
+    };
+
+    let mut cbor = serde_cbor::ser::Serializer::new(Vec::new());
+    cbor.self_describe().unwrap();
+    sig.serialize(&mut cbor).unwrap();
+    Some(cbor.into_inner())
+}
+
+fn add_signature(sigs: &mut SignatureMap, msg_hash: Hash, seed: Hash) {
```

Comment:
dito

Reply:
Yes, but the expiration could easily be a parameter. But I don't see why we would need different signature expirations? The expiration is only to give the client time to query the certificate after the update call. Whether you do that for delegations or id aliases does not really matter.

TLDR: this expiration is a property of the canister signature process not of the signed payload.

Reply:
Ok, I added `add_signature` to the ticket.